### DISCUSSION TITLE
umi max中有没有类似vite的import.meta.glob的函数，可以导入服务器端的文件，用于在前端展示文件列表。

### DISCUSSION THREAD

**Discussion by Joe-Ecust** [umijs/umi#12055] [Q&A]
想要在前端动态的展示pages文件夹下的index.tsx文件。在vite中可以使用import.meta.glob，不知道在umi中是否有类似功能的函数呢。或者有没有什么好的办法可以获取pages目录下的文件。

**Comment by fz6m** [comment]
早期 webpack 中有 `require.context` 可以一次性导入一个目录里所有的文件，但现在已经很少人这么用了，也不推荐使用，因为他具有很多不确定性，不管是导入的文件数量还是内容都是在代码预期之外的隐形内容，在 esm 静态导入为主的代码中这种行为很不利于维护（假如其他开发者参与开发，他通过搜索代码也找不到这些批量导入的外部内容来自于哪里），所以需要用到什么就显示的导入什么，如果内容非常多，手动生成导入语句也是可以的，尽量不要使用这种批量导入的语法（除非是文档类的项目，并不是以生产为主的业务）。
